pkg/build/builder/dockerutil_test.go

```diff
@@ -25,6 +29,31 @@ type methodCall struct {
 	args       []interface{}
 }
 
+func NewFakeDockerClient() *FakeDocker {
+	return &FakeDocker{}
+}
+
+func fakePushImageFunc(opts docker.PushImageOptions, auth docker.AuthConfiguration) error {
+	switch opts.Tag {
+	case "tag_test_succ_foo_bar":
+		return nil
+	case "tag_test_err_exist_foo_bar":
+		switch fooBarRunTimes {
+		case 0:
+			fooBarRunTimes++
+			return errors.New(RetriableErrors[0])
+		case 1:
+			fooBarRunTimes++
+			return errors.New(RetriableErrors[1])
+		case 2:
+			fooBarRunTimes = 0
+			return errors.New(RetriableErrors[2])
+		}
```

Comment:
if you just get check the value of fooBarRuntimes, you can get rid of this switch and just make it:
fooBarRunTimes++
return errors.New(RetriableErrors[0])